runner.py

```diff
@@ -0,0 +1,707 @@
+                test_config['target'] = target
+                test_case = TestCaseFactory.create(test_config)
+                tests.append(test_case)
+
+        if self.build:
+            for target, tests in self.tests_per_target.items():
+                target_builder = TargetBuilderFactory.create(target)
+
+                syspage = [test_case.syspage for test_case in tests]
+                syspage = list(set(syspage))
+                target_builder.build(syspage)
+
+        for target, tests in self.tests_per_target.items():
+            for test_case in tests:
+                self.runners[target].run(test_case)
+                test_case.report()
+
+        for target, tests in self.tests_per_target.items():
+            # Convert bools to int
+            for test in tests:
+                self.passed += int(test.passed())
+                self.failed += int(test.failed())
+                self.ignored += int(test.ignored())
+
+        return self.passed, self.failed, self.ignored
+
+
+def remove_prefix(string, prefix):
+    if string.startswith(prefix):
+        return string[len(prefix):]
+
+    return string
+
+
+def set_logger(level=logging.INFO):
+    root = logging.getLogger()
+    root.setLevel(logging.DEBUG)
+
+    stream_handler = logging.StreamHandler(sys.stdout)
+    stream_handler.setLevel(level)
+    formatter = logging.Formatter('%(message)s')
+    stream_handler.setFormatter(formatter)
+    root.addHandler(stream_handler)
+
+
+def args_file(arg):
+    path = pathlib.Path(arg)
+    if not path.exists():
+        print(f"Path {path} does not exist")
+        sys.exit(1)
+
+    path = path.resolve()
+    return path
+
+
+def parse_args():
+    verbosity = {
+            'debug': logging.DEBUG,
+            'info': logging.INFO,
+            'warning': logging.WARNING,
+            'error': logging.ERROR
+    }
+
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("-T", "--target",
+                        action='append', choices=TargetBuilder.TARGETS,
+                        help="Filter targets on which test will be build and ran. "
+                             "By default uses environment variable TARGET. If variable "
+                             "is not set then runs tests on all aviable targets. "
+                             "Flag can be used multiple times.")
+
+    parser.add_argument("-t", "--test",
+                        default=[], action='append', type=args_file,
+                        help="Specify directory in which test are will be searched. "
+                             "If flag is not used then runner search for tests in "
+                             "phoenix-rtos-tests dictionary. Flag can be used multiple times.")
+
+    parser.add_argument("--no-build",
+                        default=False, action='store_true',
+                        help="Omitt building step.")
+
+    parser.add_argument("-v", "--verbose",
```

Comment:
hmm, I would expect `-v` to set verbosity level to DEBUG, maybe `-l` like log-level? - or `-s[s]` for decreasing verbosity (silent)

Reply:
I changed it to `--log-level`, `-l`